STL_Extension/include/CGAL/Compact_container.h

```diff
@@ -875,6 +875,8 @@ namespace internal {
       m_ptr = nullptr;
     }
 
+    CC_iterator(pointer ptr) : m_ptr(ptr) { }
+
```

Comment:
This new constructor is the culprit for all the compilation errors.
See for example https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.3-Ic-66/Linear_cell_complex_Demo/TestReport_lrineau_ArchLinux-CXX17-Release.gz
```
In file included from /home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX17-Release/test/Linear_cell_complex_Demo/MainWindow.cpp:22:
/home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX17-Release/test/Linear_cell_complex_Demo/import_moka.h: In constructor ‘CGAL::GDart::GDart()’:
/home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX17-Release/test/Linear_cell_complex_Demo/import_moka.h:23:30: error: call of overloaded ‘CC_iterator(NULL)’ is ambiguous
   23 |   GDart() : dh(NULL), vh(NULL)
      |                              ^
In file included from /mnt/testsuite/include/CGAL/Triangulation_data_structure_2.h:33,
                 from /mnt/testsuite/include/CGAL/Triangulation_2.h:32,
                 from /mnt/testsuite/include/CGAL/Constrained_triangulation_2.h:25,
                 from /mnt/testsuite/include/CGAL/Constrained_Delaunay_triangulation_2.h:20,
                 from /home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX17-Release/test/Linear_cell_complex_Demo/typedefs.h:20,
                 from /home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX17-Release/test/Linear_cell_complex_Demo/MainWindow.h:16,
                 from /home/cgal_tester/build/src/cmake/platforms/ArchLinux-CXX17-Release/test/Linear_cell_complex_Demo/MainWindow.cpp:14:
/mnt/testsuite/include/CGAL/Compact_container.h:906:5: note: candidate: ‘CGAL::internal::CC_iterator<DSC, Const>::CC_iterator(std::nullptr_t) [with DSC = CGAL::Compact_container<CGAL::Dart<3, CGAL::CMap_linear_cell_complex_storage_1<3, 3, CGAL::Linear_cell_complex_traits<3, CGAL::Epick>, Myitems, std::allocator<int>, CGAL::Boolean_tag<false> >, CGAL::Void, CGAL::Boolean_tag<false> >, std::allocator<CGAL::Dart<3, CGAL::CMap_linear_cell_complex_storage_1<3, 3, CGAL::Linear_cell_complex_traits<3, CGAL::Epick>, Myitems, std::allocator<int>, CGAL::Boolean_tag<false> >, CGAL::Void, CGAL::Boolean_tag<false> > >, CGAL::Default, CGAL::Default>; bool Const = false; std::nullptr_t = std::nullptr_t]’
  906 |     CC_iterator (std::nullptr_t /*CGAL_assertion_code(n)*/)
      |     ^~~~~~~~~~~
/mnt/testsuite/include/CGAL/Compact_container.h:878:5: note: candidate: ‘CGAL::internal::CC_iterator<DSC, Const>::CC_iterator(CGAL::internal::CC_iterator<DSC, Const>::pointer) [with DSC = CGAL::Compact_container<CGAL::Dart<3, CGAL::CMap_linear_cell_complex_storage_1<3, 3, CGAL::Linear_cell_complex_traits<3, CGAL::Epick>, Myitems, std::allocator<int>, CGAL::Boolean_tag<false> >, CGAL::Void, CGAL::Boolean_tag<false> >, std::allocator<CGAL::Dart<3, CGAL::CMap_linear_cell_complex_storage_1<3, 3, CGAL::Linear_cell_complex_traits<3, CGAL::Epick>, Myitems, std::allocator<int>, CGAL::Boolean_tag<false> >, CGAL::Void, CGAL::Boolean_tag<false> > >, CGAL::Default, CGAL::Default>; bool Const = false; CGAL::internal::CC_iterator<DSC, Const>::pointer = CGAL::Dart<3, CGAL::CMap_linear_cell_complex_storage_1<3, 3, CGAL::Linear_cell_complex_traits<3, CGAL::Epick>, Myitems, std::allocator<int>, CGAL::Boolean_tag<false> >, CGAL::Void, CGAL::Boolean_tag<false> >*]’
  878 |     CC_iterator(pointer ptr) : m_ptr(ptr) { }
      |     ^~~~~~~~~~~
/mnt/testsuite/include/CGAL/Compact_container.h:855:9: note: candidate: ‘constexpr CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Dart<3, CGAL::CMap_linear_cell_complex_storage_1<3, 3, CGAL::Linear_cell_complex_traits<3, CGAL::Epick>, Myitems, std::allocator<int>, CGAL::Boolean_tag<false> >, CGAL::Void, CGAL::Boolean_tag<false> >, std::allocator<CGAL::Dart<3, CGAL::CMap_linear_cell_complex_storage_1<3, 3, CGAL::Linear_cell_complex_traits<3, CGAL::Epick>, Myitems, std::allocator<int>, CGAL::Boolean_tag<false> >, CGAL::Void, CGAL::Boolean_tag<false> > >, CGAL::Default, CGAL::Default>, false>::CC_iterator(const CGAL::internal::CC_iterator<CGAL::Compact_container<CGAL::Dart<3, CGAL::CMap_linear_cell_complex_storage_1<3, 3, CGAL::Linear_cell_complex_traits<3, CGAL::Epick>, Myitems, std::allocator<int>, CGAL::Boolean_tag<false> >, CGAL::Void, CGAL::Boolean_tag<false> >, std::allocator<CGAL::Dart<3, CGAL::CMap_linear_cell_complex_storage_1<3, 3, CGAL::Linear_cell_complex_traits<3, CGAL::Epick>, Myitems, std::allocator<int>, CGAL::Boolean_tag<false> >, CGAL::Void, CGAL::Boolean_tag<false> > >, CGAL::Default, CGAL::Default>, false>&)’
  855 |   class CC_iterator
      |         ^~~~~~~~~~~
```

There is already a constructor from pointers, but it is private, and take an extra `int` for disambiguation. See https://github.com/sgiraudot/cgal/blob/6bbd72cfde33f76eba3cf692ba969ad3735d802c/STL_Extension/include/CGAL/Compact_container.h#L950-L962.

Where in this PR is the new constructor needed?

Reply:
It's needed pretty much everywhere: the original implementation assumes that iterators can be created from pointers are the underlying structure is `In_place_list`. Is there any reason why this constructor should be private? The `CC_iterator` is pretty much just a pointer, and it's quite handy to be able to construct one from the other.

I could also make a named constructor that would be only used by my classes, just to be conservative with the current public/private functions.

Reply:
My guess is that was to avoid errors with implicit conversions, in the past before C++11 `explicit` constructors. I think the constructor from a pointer could be made public, but not documented, but only if it is `explicit`.